gallery/tutorials/pipeline_demo.py

```diff
@@ -0,0 +1,266 @@
+
+
+# %%
+# Noise and CTF Filters
+# ^^^^^^^^^^^^^^^^^^^^^
+# Let's start by creating noise and CTF filters. The ``operators`` package contains a collection
+# of filter classes that can be supplied to a ``Simulation``. We use ``ScalarFilter`` to create
+# Gaussian white noise and ``RadialCTFFilter`` to generate a set of CTF filters with various defocus values.
+
+# Create noise and CTF filters
+from aspire.operators import RadialCTFFilter, ScalarFilter
+
+# Gaussian noise filter.
+# Note, the value supplied to the ``ScalarFilter``, chosen based on other parameters
+# for this quick tutorial, can be changed to adjust the power of the noise.
+noise_filter = ScalarFilter(value=1e-5)
+
+# Radial CTF Filter
+defocus_min = 15000  # unit is angstroms
+defocus_max = 25000
+defocus_ct = 7
+
+ctf_filters = [
+    RadialCTFFilter(pixel_size=5, defocus=d)
+    for d in np.linspace(defocus_min, defocus_max, defocus_ct)
+]
+
+
+# %%
+# Initialize Simulation Object
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# We feed our ``Volume`` and filters into ``Simulation`` to generate the dataset of images.
+
+from aspire.source import Simulation
+
+# set parameters
+res = 41
+n_imgs = 2500
+
+# For this ``Simulation`` we set all 2D offset vectors to zero,
+# but by default offset vectors will be randomly distributed.
+src = Simulation(
+    L=res,  # resolution
+    n=n_imgs,  # number of projections
+    vols=vol,  # volume source
+    offsets=np.zeros((n_imgs, 2)),  # Default: images are randomly shifted
+    noise_filter=noise_filter,
+    unique_filters=ctf_filters,
+)
+
+
+# %%
+# Several Views of the Projection Images
+# --------------------------------------
+# We can access several views of the projection images.
+
+# with no corruption applied
+src.projections[0:10].show()
+
+# %%
+
+# with no noise corruption
+src.clean_images[0:10].show()
+
+# %%
+
+# with noise and CTF corruption
+src.images[0:10].show()
+
+
+# %%
+# CTF Correction
+# --------------
+# We apply ``phase_flip()`` to correct for CTF effects.
+
+src.phase_flip()
+src.images[0:10].show()
+
+
+# %%
+# Class Averaging
+# ---------------
+# We use ``RIRClass2D`` object to classify the images via the rotationally invariant
+# representation (RIR) algorithm. Class selection is customizable. The classification module
+# also includes a set of protocols for selecting a set of images to be used for classification.
+# Here we're using ``TopClassSelector``, which selects the first ``n_classes`` images from the source.
+
+from aspire.classification import RIRClass2D, TopClassSelector
+
+# set parameters
+n_classes = 200
+n_nbor = 6
+
+# Create a class averaging instance. Note that the ``fspca_components`` and
+# ``bispectrum_components`` were selected for this small tutorial.
+rir = RIRClass2D(
+    src,
+    fspca_components=40,
+    bispectrum_components=30,
+    n_nbor=n_nbor,
+    n_classes=n_classes,
+    selector=TopClassSelector(),
+)
```

Comment:
`    num_procs=1,  # Change to "auto" if your machine has many processors`

On my machine, this demo attempts to use multiprocessing (via ray), and this didn't seem to get along with `sphinx-gallery` (errors..).  We have had to do this for other galleries and pytest.

Reply:
Done.